### PR TITLE
Task4 taggable

### PR DESCRIPTION
**Pull request by MarioGafeed**
without Delete
Without BootStrap OR JQ

### CODE REVIEW

File: app/Http/Repositories/PostRepository.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 use App\Http\Traits\PostTrait;
 use App\Models\Post;
 use App\Models\Pcategory;
-use App\Models\Ptaq;
+// use App\Models\Ptaq;
```
Comment (mo7zayed) [inline app/Http/Repositories/PostRepository.php:9]: Remove the comment

File: app/Http/Repositories/PostRepository.php [hunk 2]
```diff
@@ -18,12 +18,11 @@ class PostRepository implements PostInterface
     use PostTrait;
     private $postModel;
     private $catModel;
-    private $taqModel;
-    public function __construct(Post $post, Pcategory $cat, Ptaq $taq)
+    // private $taqModel;
```
Comment (mo7zayed) [inline app/Http/Repositories/PostRepository.php:21]: Remove all unneeded comments

File: app/Http/Requests/VpostsRequest.php [hunk 3]
```diff
@@ -31,7 +31,7 @@ public function rules()
             'title_en'      => 'required',
             'title_ar'      => 'nullable',
             'vcat_id'       => 'required|exists:vcategories,id',
-            'vtaq_id'       => 'required|exists:vtaqs,id',
+            // 'vtaq_id'       => 'required|exists:vtaqs,id',
```
Comment (mo7zayed) [inline app/Http/Requests/VpostsRequest.php:34]: Remove the comment

File: app/Http/Repositories/PostRepository.php [hunk 4]
```diff
@@ -108,7 +105,13 @@ public function edit($id)
     {
       $pos = $this->getPostFirst($id);
       $pcats = $this->getAllpcategory();
-      $ptaq = $this->getAlltaqs();
+
+      $tagNames = $pos->tags->pluck('name');
+      foreach ($tagNames as $key => $tagName) {
+        $results[] = $tagName;
+      }
+      $tags = implode(',', $results);
```
Comment (mo7zayed) [inline app/Http/Repositories/PostRepository.php:113]:
```php
$tags = $pos->tags->pluck('name')->implode(', ')->toArray();
```
Isn't it simpler & cleaner

File: app/Http/Repositories/PostRepository.php [hunk 5]
```diff
@@ -73,11 +71,10 @@ public function store($request)
...
         $pos->addMediaFromRequest('image')->toMediaCollection();
       }
-      
-      if ($requestAll['ptaq_id'])
-      {
-        $pos->ptaqs()->attach($requestAll['ptaq_id']);
-      }
+
+      $tags = explode(',' ,$request->tags);
+      $vpos->attachTags($tags);
```
Comment (mo7zayed) [inline app/Http/Repositories/PostRepository.php:76]: No variable named `$vpos`!

File: app/Http/Repositories/VpostRepository.php [hunk 6]
```diff
@@ -102,16 +95,22 @@ public function edit($id)
     {
       $vpos = $this->getVpostFirst($id);
       $vcat = $this->getAllvcategory();
-      $vtaq = $this->getAlltaqs();
+
+      $tagNames = $vpos->tags->pluck('name');
+      foreach ($tagNames as $key => $tagName) {
+        $results[] = $tagName;
+      }
+      $tags = implode(',', $results);
```
Comment (mo7zayed) [inline app/Http/Repositories/VpostRepository.php:103]: the same https://github.com/MarioGafeed/DoctorsCamp/pull/6/files#r771880830

File: app/Http/Requests/VpostsRequest.php (hunk 3 above)
Comment (mo7zayed) [inline app/Http/Requests/VpostsRequest.php:34]: Remove the comment

Comment (MarioGafeed) [reply app/Http/Requests/VpostsRequest.php:34]: Done

File: app/Http/Repositories/VpostRepository.php (hunk 6 above)
Comment (MarioGafeed) [reply app/Http/Repositories/VpostRepository.php:103]: Done

File: app/Http/Repositories/PostRepository.php (hunk 5 above)
Comment (MarioGafeed) [reply app/Http/Repositories/PostRepository.php:76]: Fix

File: app/Http/Repositories/PostRepository.php (hunk 4 above)
Comment (MarioGafeed) [reply app/Http/Repositories/PostRepository.php:113]: Done

File: app/Http/Requests/VpostsRequest.php (hunk 3 above)
Comment (MarioGafeed) [reply app/Http/Requests/VpostsRequest.php:34]: Done

File: app/Http/Repositories/PostRepository.php (hunk 2 above)
Comment (MarioGafeed) [reply app/Http/Repositories/PostRepository.php:21]: Done

File: app/Http/Repositories/PostRepository.php (hunk 1 above)
Comment (MarioGafeed) [reply app/Http/Repositories/PostRepository.php:9]: Done

**Comment by MarioGafeed** [comment]
All done, Thanks Zayed..

**Pull request merged**